includes/Modules/Analytics_4.php

```diff
@@ -1372,6 +1406,104 @@ function ( $order_def ) {
 		return $results;
 	}
 
+	/**
+	 * Parses the report response.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request                                   $data     Data request object.
+	 * @param Google_Service_AnalyticsData_RunReportResponse $response Request response.
+	 * @return mixed Parsed response data on success, or WP_Error on failure.
+	 */
+	protected function parse_report_response( Data_Request $data, $response ) {
+		// Return early if the response is not of the expected type.
+		if ( ! $response instanceof Google_Service_AnalyticsData_RunReportResponse ) {
+			return $response;
+		}
+
+		// Get report dimensions and return early if there is either more than one dimension or
+		// the only dimension is not "date".
+		$dimensions = $this->parse_reporting_dimensions( $data );
+		if ( count( $dimensions ) !== 1 || $dimensions[0]->getName() !== 'date' ) {
+			return $response;
+		}
+
+		// Get date ranges and return early if there is no date ranges for this report.
+		$date_ranges = $this->parse_reporting_dateranges( $data );
+		if ( empty( $date_ranges ) ) {
+			return $response;
+		}
+
+		$rows    = $response->getRows();
+		$metrics = $response->getMetricHeaders();
+
+		foreach ( $date_ranges as $date_range ) {
+			$start = strtotime( $date_range->getStartDate() );
+			$end   = strtotime( $date_range->getEndDate() );
+
+			// Skip this date range if either start date or end date is corrupted.
+			if ( ! $start || ! $end ) {
+				continue;
+			}
+
+			// Loop through all days in the date range and check if there is a metric value
+			// for it. If the metric value is missing, we will need to add one with a zero value.
+			$now = $start;
+			do {
+				$current_date = gmdate( 'Ymd', $now );
+
+				// Search for the current date in the response rows.
+				$found_dimenension = false;
+				foreach ( $rows as $row ) {
+					$dimention_values = $row->getDimensionValues();
+					if ( $dimention_values[0]->getValue() === $current_date ) {
+						$found_dimenension = true;
+						break;
+					}
+				}
+
+				// If the current date is not found, add the new row to the existing rows.
+				if ( ! $found_dimenension ) {
+					$dimension_value = new Google_Service_AnalyticsData_DimensionValue();
+					$dimension_value->setValue( $current_date );
+
+					$metric_values = array();
+					foreach ( $metrics as $metric ) {
+						$metric_value = new Google_Service_AnalyticsData_MetricValue();
+
+						switch ( $metric->getType() ) {
+							case 'TYPE_INTEGER':
+							case 'TYPE_FLOAT':
+							case 'TYPE_CURRENCY':
+								$metric_value->setValue( 0 );
```

Comment:
> Looking into the response schema a bit more closely it's evident that the value is a string for all metric types.

Honestly, it doesn't matter for us as long as we get `0` on the javascript side, but anyway updated to be `'0'`.

> ... and creating a followup issue to get the detail on `METRIC_TYPE_UNSPECIFIED`.

No need to create a new issue for it. It's useless in our case, we should care about currency and numeric types only, everything else will never be used for date data.

Reply:
I think it's preferable to keep the data format consistent with the GA4 result, though - we should stick to the schema, to avoid unexpected surprises.

I've already raised a separate comment about the data types - I think we should just set a zero for all types, I don't see why we need to differentiate the types here.